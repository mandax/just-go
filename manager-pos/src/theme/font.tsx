import theme from "./index";
import { rem } from "./units";

export type FontConstructor =
  (fontSize?: number, fontFamily?: string) => React.CSSProperties;

export const fontBase = (
  fontWeight: number = theme.FONT_DEFAULT_WEIGHT,
): FontConstructor => (
  size: number = theme.FONT_DEFAULT_SIZE,
  fontFamily: string = theme.FONT_PRIMARY
): React.CSSProperties => ({
  fontFamily,
  fontWeight,
  fontSize: rem(size)
});

export const font: FontConstructor = fontBase(theme.FONT_DEFAULT_WEIGHT);

export const fontMedium: FontConstructor = fontBase(theme.FONT_MEDIUM_WEIGHT);

export const fontStrong: FontConstructor = fontBase(theme.FONT_STRONG_WEIGHT);

export const capitalize = (str: string): string => str[0].toUpperCase() + str.slice(1);

export const limitChar = (text: string, limit: number): string => {
  if (text.length > limit) {
    text = `${text.substring(0, limit)}...`;
  }
  return text;
}

export const colorByState = (state: boolean): React.CSSProperties => ({
  color: state ? theme.COLOR_BRAND : theme.COLOR_PRIMARY
});

export const titleSpacing = (
  marginTop: number = theme.TITLE_SPACING_TOP,
  marginBottom: number = theme.TITLE_SPACING_BOTTOM
): React.CSSProperties => ({
  marginTop: rem(marginTop),
  marginBottom: rem(marginBottom)
});