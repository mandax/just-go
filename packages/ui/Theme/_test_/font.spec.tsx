import * as font from "../font";

describe('helper font', () => {
    
  it('should generate a default font config', () => {
    expect(font.fontBase()()).toMatchSnapshot()
  })
  
})