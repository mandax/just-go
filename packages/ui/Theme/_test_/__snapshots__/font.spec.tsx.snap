// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helper font should generate a default font config 1`] = `
Object {
  "fontFamily": "Manjari",
  "fontSize": "1rem",
  "fontWeight": 400,
}
`;
